daemon.go

```diff
@@ -14,156 +14,53 @@
 package main
 
 import (
-	"time"
-
-	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 // Config section
 
-// daemon configuration
-type daemonConfig struct {
-	serverpollInterval time.Duration
-	serverURL          string
-	deviceID           string
-}
-
-// Daemon section
-
 type menderDaemon struct {
-	Updater
-	UInstallCommitRebooter
-	Controler
-	config      daemonConfig
+	mender      Controler
 	stopChannel chan (bool)
 	stop        bool
 }
 
-func NewDaemon(client Updater, device UInstallCommitRebooter,
-	mender Controler) *menderDaemon {
-
-	config := mender.GetDaemonConfig()
+func NewDaemon(mender Controler) *menderDaemon {
 
 	daemon := menderDaemon{
-		Updater:                client,
-		UInstallCommitRebooter: device,
-		Controler:              mender,
-		config:                 config,
-		stopChannel:            make(chan bool),
+		mender:      mender,
+		stopChannel: make(chan bool),
 	}
 	return &daemon
 }
 
-func (daemon menderDaemon) StopDaemon() {
-	daemon.stopChannel <- true
+func (d menderDaemon) StopDaemon() {
```

Comment:
Are we using this anywhere right now? I've seen you rather are setting error and checking this in main daemon loop. If so maybe we can get rid of it and remove `stopChannel` from `menderDaemon` as well.